mqtt/mqtt-edgehub/src/auth/authorization/edgehub.rs

```diff
@@ -119,6 +121,38 @@ impl EdgeHubAuthorizer {
             .iter()
             .any(|allowed_topic| topic.starts_with(allowed_topic))
     }
+
+    fn check_authorized_cache(&self, client_id: &ClientId, topic: &str) -> bool {
+        let on_behalf_of_id: &str = match topic.split('/').collect::<Vec<&str>>().get(1) {
+            Some(id) => {
+                id
+            }
+            None => {
+                return false
+            }
+        };
```

Comment:
I would extract this into a separate func `get_on_behalf_of_id()`. That logic might change later.